pkgpanda/util.py

```diff
@@ -150,6 +151,38 @@ def get_requests_retry_session(max_retries=4, backoff_factor=1, status_forcelist
     return session
 
 
+def stream_remote_file_with_retries(out_filename, url, retries=4):
+    def cleanup():
+        try:
+            os.remove(out_filename)
+        except Exception:
+            pass
+
+    while True:
+        with open(out_filename, "w+b") as f:
+            r = get_requests_retry_session().get(url, stream=True)
+            if r.status_code == 301:
```

Comment:
Why do you care about redirects? It seems reasonable that the CDN / S3 would redirect you to a specific download URL.

Reply:
As I wrote in Slack: I — personally — don't care 😄 I've taken this from the existing (untested) code that was already there, presumably for years. And when working with legacy code that doesn't have tests to confirm something. I tread lightly and only really change what needs to be changed. Or, in other words: if I don't know why code is there and don't know why it _shouldn't_ be there I'll leave it alone — for now, because I don't want to break something that I didn't know existed.

Reply:
SGTM